pilot/pkg/xds/eds_test.go

```diff
@@ -437,6 +437,85 @@ func TestDeleteService(t *testing.T) {
 	}
 }
 
+func TestUpdateServiceAccount(t *testing.T) {
+	cluster1Endppoints := []*model.IstioEndpoint{
+		{Address: "10.172.0.1", ServiceAccount: "sa1"},
+		{Address: "10.172.0.2", ServiceAccount: "sa-vm1"},
+	}
+
+	testCases := []struct {
+		name      string
+		clusterID string
+		endpoints []*model.IstioEndpoint
+		expect    bool
+	}{
+		{
+			name:      "added new endpoint",
+			clusterID: "c1",
+			endpoints: append(cluster1Endppoints, &model.IstioEndpoint{Address: "10.172.0.3", ServiceAccount: "sa1"}),
+			expect:    false,
+		},
+		{
+			name:      "added new sa",
+			clusterID: "c1",
+			endpoints: append(cluster1Endppoints, &model.IstioEndpoint{Address: "10.172.0.3", ServiceAccount: "sa2"}),
+			expect:    true,
+		},
+		{
+			name:      "updated endpoints address",
+			clusterID: "c1",
+			endpoints: []*model.IstioEndpoint{
+				{Address: "10.172.0.5", ServiceAccount: "sa1"},
+				{Address: "10.172.0.2", ServiceAccount: "sa-vm1"},
+			},
+			expect: false,
+		},
+		{
+			name:      "deleted one endpoint with duplicate sa",
+			clusterID: "c1",
+			endpoints: []*model.IstioEndpoint{
+				{Address: "10.172.0.1", ServiceAccount: "sa1"},
+			},
+			expect: true,
```

Comment:
if its duplicate why do we trigger update?

Reply:
updated